### PR TITLE
python3Packages.terraformpy

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Starting to play around with terraformpy wanted to contribute the packaged libraries.

###### Things done

 - python3Packages.schematics
 - python3Packages.terraformpy

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 148084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python38Packages.mo_installer</li>
    <li>python38Packages.schematics</li>
    <li>python38Packages.terraformpy</li>
    <li>python39Packages.mo_installer</li>
    <li>python39Packages.schematics</li>
    <li>python39Packages.terraformpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/mo_installer/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  # contains no tests
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/mo_installer/default.nix:24]: looks like they have tags https://github.com/s-ball/mo_installer/tags and test in their source, do you mind trying to build from the repo?

Also, the repo looks to have 0 activity in 3  years.

Comment (costrouc) [reply pkgs/development/python-modules/mo_installer/default.nix]: @jonringer I resolved this by removing the mo_installer package entirely. The tests for that package seem to fail and mo_installer was only a build time dependency in schematics that does not seem to be used.

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 148084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.schematics</li>
    <li>python38Packages.terraformpy</li>
    <li>python39Packages.schematics</li>
    <li>python39Packages.terraformpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/mo_installer/default.nix (hunk 1 above)
Comment (costrouc) [reply pkgs/development/python-modules/mo_installer/default.nix]: Also I was not really happy either adding a package that was not maintained like that. Thanks for the review!

**Comment by costrouc** [comment]
Rebased ready for merge.

File: pkgs/development/python-modules/schematics/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-jclKcX/4QbRCuWKdKq97Wo3q10Rbkt7/R8AJQTwnwVk=";
+  };
+
+  buildInputs = [
```
Comment (jonringer) [inline pkgs/development/python-modules/schematics/default.nix:23]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/development/python-modules/schematics/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  # mo_installer is not maintained and not required for running this
+  # package
+  postPatch = ''
+    substituteInPlace setup.py --replace "'mo_installer'," ""
+  '';
+
+  checkPhase = ''
+    pytest
+  '';
+
```
Comment (jonringer) [inline pkgs/development/python-modules/schematics/default.nix:44]:
```suggestion
```

File: pkgs/development/python-modules/schematics/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-jclKcX/4QbRCuWKdKq97Wo3q10Rbkt7/R8AJQTwnwVk=";
+  };
+
+  buildInputs = [
+    pytest-runner
+  ];
+
+  checkInputs = [
+    pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/schematics/default.nix:28]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/schematics/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{ lib,
+  fetchFromGitHub,
+  buildPythonPackage,
+  pytest-runner,
+  pytest,
```
Comment (jonringer) [inline pkgs/development/python-modules/schematics/default.nix:5]:
```suggestion
  pytestCheckHook,
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 148084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.schematics</li>
    <li>python38Packages.terraformpy</li>
    <li>python39Packages.schematics</li>
    <li>python39Packages.terraformpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/schematics/default.nix (hunk 5 above)
Comment (costrouc) [reply pkgs/development/python-modules/schematics/default.nix]: added

File: pkgs/development/python-modules/schematics/default.nix (hunk 4 above)
Comment (costrouc) [reply pkgs/development/python-modules/schematics/default.nix]: added

File: pkgs/development/python-modules/schematics/default.nix (hunk 3 above)
Comment (costrouc) [reply pkgs/development/python-modules/schematics/default.nix]: added

File: pkgs/development/python-modules/schematics/default.nix (hunk 2 above)
Comment (costrouc) [reply pkgs/development/python-modules/schematics/default.nix]: added

**Comment by costrouc** [comment]
Added the pytestCheckHook

**Comment by costrouc** [comment]
Ready for merge

**Comment by onny** [comment]
Please rebase

**Pull request closed without merging**